### PR TITLE
Fixed behaviour of checkboxes in legend pane

### LINKED ISSUE #21: Line chart's Legend tab does not behave correctly when split on filtered dimension
Opening the following [link](http://localhost:9090/#wiki/line-chart/2/EQUQLgxg9AqgKgYWAGgN7APYAdgC5gCGAdgCYqZYCmATsWbutnpgG40A2BOyFNdejHPmqUAZuSIEAtpWZgAljOABfHpQAeWEQGdt8jEQEU5iygCViAc1k9sfUkabCxE6bPxSVPEgFdaCg2YABQBGABFybTBKIQBaEOVVYA0tSl19QwZjfAw2ak5uXloHLKdgEXEeSSV8CAwfIjBqAE8ASW0MBAwSWSSUnT1A0qFgdnlo2nZyFgJ2H3d0bUowOGaqZgBlODNWgDkAcXJKdkoZRu08AG1gGA3yVrhyfYAhcgAxM3IzGHIAKSDyGEQMAALpJMBrdzADYgR6JVSXdD9NKDTKCZgVVw1YB1BpNNodLo9LzADrUMAAQQgATR2VJGHJR00AwyjhGmKqbmYPXYYAIJJI8hE1NZ+B62gglFI8iIlhUqiRzJRovRzkqwGqUIUSiSACMfBAANbLKk0tkmGTPA3GsDkXz+UXAIJwEIACRJZMpIqGqvpjLUSvSPrpHI1XPw2t63iFlG9mUIEqlgtl8pByCa828xz5KGuuMaoOQefqjRa7QwhaIPnY7DQwEFZ1RzHz+PL5DGUnGeAArMogA===) with Turnilo started in wikipedia example mode presents an hourly line chart filtered for 7 "Country ISOs" and split by "Country ISO" with Legend tile on the right.

There are following issues:
- Legend contains only 5 elements instead of 7 with first 5 selected
- deselecting element in Legend removes corresponding value from filter but it should only hide the serie from line chart

### PR DESCRIPTION
**Pull request by rafalnowak**
Fixed behaviour of checkboxes in legend pane - now they are not removing items from filter.

Fixes #21

### CODE REVIEW

File: src/client/components/dimension-tile/dimension-tile.tsx [hunk 1]
```diff
@@ -493,73 +461,110 @@ export class DimensionTile extends React.Component<DimensionTileProps, Dimension
...
+      return rowsHeightWithPaddingAndTitle;
     }
+  }
 
-    maxHeight += PIN_PADDING_BOTTOM;
+  render() {
+    const { essence, dimension, colors, onClose } = this.props;
+    const { loading, dataset, error, showSearch, unfolded, foldable, fetchQueued, searchText, filterMode } = this.state;
+
+    const continuous = dimension.isContinuous();
```
Comment (mkuthan) [inline src/client/components/dimension-tile/dimension-tile.tsx]: isContinuous?

File: src/client/components/dimension-tile/dimension-tile.tsx [hunk 2]
```diff
@@ -493,73 +461,110 @@ export class DimensionTile extends React.Component<DimensionTileProps, Dimension
...
+    const rowData = this.prepareRowsData();
```
Comment (mkuthan) [inline src/client/components/dimension-tile/dimension-tile.tsx]: rowsData?

File: src/client/components/dimension-tile/dimension-tile.tsx [hunk 3]
```diff
@@ -493,73 +461,110 @@ export class DimensionTile extends React.Component<DimensionTileProps, Dimension
...
+    }
+  }
 
-        if (segmentValue instanceof TimeRange) {
-          segmentValueStr = formatTimeBasedOnGranularity(segmentValue, (selectedGranularity as TimeBucketExpression).duration, essence.timezone, getLocale());
-        } else if (segmentValue instanceof NumberRange) {
-          segmentValueStr = formatNumberRange(segmentValue);
-        }
+  private prepareColorsValues(colors: Colors, dimension: Dimension, rowData: Datum[]): string[] {
+    let colorValues: string[] = null;
```
Comment (mkuthan) [inline src/client/components/dimension-tile/dimension-tile.tsx]: could you get rid of mutable variable colorValues?

File: src/client/components/dimension-tile/dimension-tile.tsx [hunk 4]
```diff
@@ -493,73 +461,110 @@ export class DimensionTile extends React.Component<DimensionTileProps, Dimension
...
+  private prepareRows(rowData: Datum[], continuous: boolean): Array<JSX.Element> {
+    const { essence, dimension, sortOn, colors } = this.props;
+    const { searchText, selectedGranularity, filterMode } = this.state;
+
+    const measure = sortOn.measure;
+    const measureName = measure ? measure.name : null;
+    const formatter = measure ? formatterFromData(rowData.map(d => d[measureName] as number), measure.getFormat()) : null;
+    const colorValues = this.prepareColorsValues(colors, dimension, rowData);
+    const filterSet = essence.filter.getLiteralSet(dimension.expression);
+    const excluded = filterMode === Filter.EXCLUDED;
```
Comment (mkuthan) [inline src/client/components/dimension-tile/dimension-tile.tsx]: isExcluded?

File: src/client/components/dimension-tile/dimension-tile.tsx [hunk 5]
```diff
@@ -493,73 +461,110 @@ export class DimensionTile extends React.Component<DimensionTileProps, Dimension
...
 
-        let measureValueElement: JSX.Element = null;
-        if (measure) {
-          measureValueElement = <div className="measure-value">{formatter(d[measureName] as number)}</div>;
+    return rowData.map((datum, i) => {
+      const segmentValue = datum[dimension.name];
+      let segmentValueStr = String(segmentValue);
```
Comment (mkuthan) [inline src/client/components/dimension-tile/dimension-tile.tsx]: move segmentValueStr to the block where variable is used

File: src/client/components/dimension-tile/dimension-tile.tsx [hunk 6]
```diff
@@ -493,73 +461,110 @@ export class DimensionTile extends React.Component<DimensionTileProps, Dimension
...
+      return [];
+    }
+  }
 
-        if (segmentValue instanceof TimeRange) {
-          segmentValueStr = formatTimeBasedOnGranularity(segmentValue, (selectedGranularity as TimeBucketExpression).duration, essence.timezone, getLocale());
-        } else if (segmentValue instanceof NumberRange) {
-          segmentValueStr = formatNumberRange(segmentValue);
-        }
+  private prepareColorsValues(colors: Colors, dimension: Dimension, rowData: Datum[]): string[] {
```
Comment (piter75) [inline src/client/components/dimension-tile/dimension-tile.tsx]: prepareColorValues?

**Pull request merged**